multiqc/utils/search_patterns.yaml

```diff
@@ -396,6 +396,12 @@ ngsderive/instrument:
 ngsderive/readlen:
   contents_re: "(.*)File(.*)Evidence(.*)MajorityPctDetected(.*)ConsensusReadLength"
   num_lines: 1
+ngsderive/encoding:
+  contents_re: "(.*)File(.*)Evidence(.*)ProbableEncoding"
```

Comment:
Could you explain why this regular expression search is needed? It's a super slow method for finding matches, especially when you're matching any number of any characters so many times. For example, another tool could generate a huge minified JSON file on one line and this regex can take minutes to run.

Can we not just do `contents: "File	Evidence	ProbableEncoding"`? Or even `contents: "ProbableEncoding"`

Remember that the module itself can discard false-positives when it runs, so this match doesn't have to be 100% perfect. As this will run on every file for every user it does need to be fast though.

Phil

Reply:
I should add that the reason I make this comment is that I was just doing the final checks before pushing a v1.10 release and I found that my final big test was going super slow / hanging indefinitely. After a little debugging (prompting an even nicer new progress bar tweak) I found that it was the `ngsderive` search patterns hanging on test data for Atropos (whose PR has been open for years but never merged). I'm not entirely sure why, but switching for a simpler search pattern fixed it.

By fixed, I mean that after the change in 6263373c86a94eaa19c832c1c0a3a27326aaf469 the  CI test run has gone from ~40 minutes total to ~3 minutes 😅 So note that `master` now has simpler search patterns for the other files, which may have broken things for you (though the report from the test data seems identical).

This is why I run tests before pushing releases...! 😀

Reply:
Ping @claymcleod @a-frantz - if you could take at this ASAP to check that the new patterns are ok (or suggest an alternative) that would be fab. It's the only thing that the release is waiting for now.

I'll have a look at this PR for you now and try to sneak that in as well as a thankyou-in-advance 😄